rclcpp_lifecycle/include/rclcpp_lifecycle/lifecycle_node.hpp

```diff
@@ -12,6 +12,22 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+/** \mainpage rclcpp_lifecycle: Package containing a prototype for lifecycle implementation.
+ *
+ * - Lifecycle states: Define the State class.
+ *   - rclcpp_lifecycle/state.hpp
+ * - Lifecycle transitions Define the Transition class.
+ *   - rclcpp_lifecycle/transition.hpp
+ * - Lifecycle publisher A managed node can then deactivate or activate the publishing.
```

Comment:
I see you got this from lifecycle_publisher.hpp itself. It looks like the docblocks need to be cleaned up. Is that something you're also able to take care of?

Otherwise, a better brief description might be, "An optional interface class for life cycle node implementations."

Reply:
This description of lifecycle publisher still needs to be updated as it's not terribly clear.